### PR TITLE
[Merged by Bors] - Use mirrord for exec

### PR DESCRIPTION
**Pull request by aviramha**

### CODE REVIEW

File: mirrord/cli/src/execution.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+impl MirrordExecution {
+    pub(crate) async fn start<P>(config: &LayerConfig, progress: &P) -> Result<Self>
+    where
+        P: Progress + Send + Sync,
+    {
+        let lib_path = extract_library(None, progress, true)?;
+        let mut env_vars = HashMap::new();
+        let (connect_info, mut connection) = create_and_connect(config, progress).await?;
+        let (env_vars_filter, env_vars_select) = match (
```
[the remark this answers is not in the document]
Comment (aviramha) [reply mirrord/cli/src/execution.rs]: done

File: mirrord/cli/src/execution.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        // If already exists, we append.
+        if let Ok(v) = std::env::var(INJECTION_ENV_VAR) {
+            env_vars.insert(INJECTION_ENV_VAR.to_string(), format!("{}:{}", v, lib_path))
+        } else {
+            env_vars.insert(INJECTION_ENV_VAR.to_string(), lib_path)
+        };
+
+        match &connect_info {
+            AgentConnectInfo::DirectKubernetes(name, port) => {
+                env_vars.insert("MIRRORD_CONNECT_AGENT".to_string(), name.to_string());
```
[the remark this answers is not in the document]
Comment (aviramha) [reply mirrord/cli/src/execution.rs:104]: done

**Review by eyalb181 — APPROVED** [review]

**Pull request closed without merging**